### PR TITLE
Bridges for Mamba-based models

### PR DESCRIPTION
**Pull request by maanug-nv**
Add two bridges for Mamba-based models:

MambaBridge which supports any `MambaForCausalLM` including:
- state-spaces/mamba-130m-hf
- state-spaces/mamba-370m-hf
- state-spaces/mamba-790m-hf
- state-spaces/mamba-1.4b-hf
- state-spaces/mamba-2.8b-hf

NemotronHBridge which supports any `NemotronHForCausalLM` including:
- nvidia/Nemotron-H-8B-Base-8K
- nvidia/Nemotron-H-47B-Base-8K
- nvidia/Nemotron-H-56B-Base-8K
- nvidia/NVIDIA-Nemotron-Nano-9B-v2
- nvidia/NVIDIA-Nemotron-Nano-12B-v2

### CODE REVIEW

File: src/megatron/bridge/models/mamba/mamba_bridge.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        megatron_module: Optional[nn.Module],
+    ) -> dict[str, torch.Tensor]:
+        """Prune padding from weight in vocab size dimension, if vocab size is accessible."""
+        mapping = super().megatron_to_hf(megatron_weights, megatron_module)
+
+        if megatron_module is not None:
+            weight = mapping[str(self.hf_param)]
+            mapping[str(self.hf_param)] = weight[: megatron_module.vocab_size, :]
+
+        return mapping
```
Comment (ananthsub) [inline src/megatron/bridge/models/mamba/mamba_bridge.py]: should this be a generic transform made available here? https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/src/megatron/bridge/models/conversion/param_mapping.py

Comment (maanug-nv) [reply src/megatron/bridge/models/mamba/mamba_bridge.py]: Was discussing this with @yaoyu-33 . In Nemo 2.0, this pruning is only done for the NemotronH exporter: https://github.com/NVIDIA-NeMo/NeMo/blob/main/nemo/collections/llm/gpt/model/ssm.py#L817-L850

I'm not sure why this wasn't needed for other models.
I think @JRD971000 mentioned that the NemotronH checkpoints were saved with vocab padding so this was a choice made when writing exporters specifically for NemotronH.

**Comment by maanug-nv** [comment]
Needs #440 to be merged first, for HF remote model support.

**Comment by maanug-nv** [comment]
Closes #500 and #149

File: src/megatron/bridge/models/mamba/nemotron_h_bridge.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        mapping_list.extend(
+            [
+                # QKV: Combine separate Q, K, V matrices into single QKV matrix
+                QKVMapping(
+                    megatron_param="decoder.layers.*.self_attention.linear_qkv.weight",
+                    q="backbone.layers.*.mixer.q_proj.weight",
+                    k="backbone.layers.*.mixer.k_proj.weight",
+                    v="backbone.layers.*.mixer.v_proj.weight",
+                ),
+                AutoMapping(megatron_param="embedding.word_embeddings.weight", hf_param="backbone.embeddings.weight"),
```
Comment (yaoyu-33) [inline src/megatron/bridge/models/mamba/nemotron_h_bridge.py]: if it's auto mapping just put in the dict above?

Comment (maanug-nv) [reply src/megatron/bridge/models/mamba/nemotron_h_bridge.py]: cleaned this up.
